ludwig/api.py

```diff
@@ -293,16 +293,16 @@ def __init__(
             config_dict = copy.deepcopy(config)
             self.config_fp = None
 
-        self.config_dict = config_dict
+        self._user_config = config_dict
```

Comment:
I like this idea of more formally distinguishing the user config from fully filled out Ludwig configs with all defaults in them.

A private variable seems appropriate, though in the future it might make sense to expose a way to fetch this from the API?

Reply:
Yeah, I agree we may wish to do that in a follow-up.